docs/event-processing/idempotent-writer.md

```diff
@@ -0,0 +1,36 @@
+---
+seo:
+  title: Idempotent Writer
+  description: An Idempotent Writer produces an Event to an Event Streaming Platform exactly once.
+---
+
+# Idempotent Writer
+A writer produces [Events](../event/event.md) that are written into an [Event Stream](../event-stream/event-stream.md), and under stable conditions, each Event is recorded only once.
+However, in the case of an operational failure or a brief network outage, an [Event Source](../event-source/event-source.md) may need to retry writes. This may result in multiple copies of the same Event ending up in the Event Stream, as the first write may have actually succeeded and the client simply did not receive the response. This type of duplication is one of the perils of distributed systems.
+
+## Problem
+How can an [Event Streaming Platform](../event-stream/event-streaming-platform.md) ensure that an Event Source does not write the same Event more than once?
+
+## Solution
+![idempotent-writer](../img/idempotent-writer.svg)
+
+Generally speaking, this can be addressed by native support for idempotent clients.
+This means that a writer may try to produce an Event more than once, but the Event Streaming Platform detects and discards duplicate write attempts for the same Event. The resulting Events in the Event Stream are unique, ensuring that a consumer's computed results remain accurate.
+
+## Implementation
+To make an Apache Kafka® producer idempotent, configure your producer with the following setting:
+
+```
+enable.idempotence=true
+```
+
+The Kafka producer tags each batch of Events that it sends to the Kafka cluster with a sequence number. The broker uses this sequence number to enforce deduplication of Events sent from this specific producer. Each batch's sequence number is persisted to the replicated log, so that even if the leader broker fails, any new leader will also know if a given batch is a duplicate.
```

Comment:
```suggestion
The Kafka producer tags each batch of Events that it sends to the Kafka cluster with a sequence number. Brokers in the cluster use this sequence number to enforce deduplication of Events sent from this specific producer. Each batch's sequence number is persisted to the replicated log, so that even if the leader broker fails, any new leader will also know if a given batch is a duplicate.
```

Reply:
> Each batch's sequence number is persisted to the replicated log, so that even if the leader broker fails, ...

The reader will not understand what "the replicated log" refers to. Same for "leader broker". Either explain with a few words, or add links to learn more.

Reply:
Updated in latest commit